pkg/apis/pingcap/v1alpha1/tidb_config.go

```diff
@@ -103,6 +103,27 @@ type TiDBConfig struct {
 	SplitRegionMaxNum *uint64 `toml:"split-region-max-num,omitempty" json:"split-region-max-num,omitempty"`
 	// +optional
 	StmtSummary *StmtSummary `toml:"stmt-summary,omitempty" json:"stmt-summary,omitempty"`
+	// RepairMode indicates that the TiDB is in the repair mode for table meta.
+	// +optional
+	RepairMode bool `toml:"repair-mode" json:"repair-mode,omitempty"`
+	// +optional
+	RepairTableList []string `toml:"repair-table-list" json:"repair-table-list,omitempty"`
+	// IsolationRead indicates that the TiDB reads data from which isolation level(engine and label).
+	// +optional
+	IsolationRead IsolationRead `toml:"isolation-read" json:"isolation-read,omitempty"`
+	// MaxServerConnections is the maximum permitted number of simultaneous client connections.
+	// +optional
+	MaxServerConnections uint32 `toml:"max-server-connections" json:"max-server-connections,omitempty"`
+	// NewCollationsEnabledOnFirstBootstrap indicates if the new collations are enabled, it effects only when a TiDB cluster bootstrapped on the first time.
+	// +optional
+	NewCollationsEnabledOnFirstBootstrap bool `toml:"new_collations_enabled_on_first_bootstrap" json:"new_collations_enabled_on_first_bootstrap,omitempty"`
+	// Experimental contains parameters for experimental features.
+	// +optional
+	Experimental Experimental `toml:"experimental" json:"experimental,omitempty"`
+	// EnableDynamicConfig enables the TiDB to fetch configs from PD and update itself during runtime.
+	// see https://github.com/pingcap/tidb/pull/13660 for more details.
+	// +optional
+	EnableDynamicConfig bool `toml:"enable-dynamic-config" json:"enable-dynamic-config,omitempty"`
```

Comment:
add tidb version requirements, links to configuration docs in comments? these comments can be our docs for API types.

Reply:
we can generate API docs for our CR types like this: https://cert-manager.io/docs/reference/api-docs/ .

Reply:
good catch

Reply:
Currently docs team is now working on marked big version for the configuration. Do we need to pend this issue or we merge this pr and mark the big version unit their work is done?

Reply:
After discussion from docs team, I think we could pend this issue and wait their version makring work done.

Reply:
currently the newly 3.1.0 added configuration is makred in note.